### ISSUE TITLE
[ISSUE] Issue with `databricks_permission` silently removes "tokens" permissions

### ISSUE THREAD

**Issue by yegorski** [databricks/terraform-provider-databricks#2431]
👋 I looked around at others issues like https://github.com/databricks/terraform-provider-databricks/issues/2268, https://github.com/databricks/terraform-provider-databricks/issues/2131, https://github.com/databricks/terraform-provider-databricks/issues/2104, and https://github.com/databricks/terraform-provider-databricks/issues/1976 but they don't quite describe my issue.

My issue is that the [databricks_permissions](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/permissions) resource removed all tokens when adding a new one when it is invoked with `authorization = "tokens"`. The documentation warns about this behavior:

```md
Configuring this resource for an object will OVERWRITE any existing permissions of the same type unless imported, and changes made outside of Terraform will be reset unless the changes are also reflected in the configuration.
```

But it would be great if the Databricks Terraform provider (we tested on 1.18.0) makes an extra DB API call to check and warn when removing tokens. Right now it silently removes previous token grants.

### Configuration

For example we were unaware that we managing token permissions in 2 TF repos. This one:

```hcl
resource "databricks_permissions" "token_permissions" {
  authorization = "tokens"

  access_control {
    service_principal_name = databricks_service_principal.service_account.application_id
    permission_level       = "CAN_USE"
  }
}
```

overrode this one:

```hcl
resource "databricks_permissions" "token_permissions" {
  authorization = "tokens"

  access_control {
    group_name       = data.databricks_group.all_users.display_name
    permission_level = "CAN_USE"
  }
}
```

And removed token access for all our users❗ 

In the end we figured it out and combined them:

```hcl
resource "databricks_permissions" "token_permissions" {
  authorization = "tokens"

  access_control {
    group_name       = data.databricks_group.all_users.display_name
    permission_level = "CAN_USE"
  }

  access_control {
    service_principal_name = databricks_service_principal.service_account.application_id
    permission_level       = "CAN_USE"
  }
}
```

### Expected Behavior

`databricks_permission` resource warns when there are existing token permissions about to be removed.

### Actual Behavior

`databricks_permission` resource silently removes other token permissions.

### Steps to Reproduce

See `Configuration` section above.

**Comment by nkvuong** [comment]
There is a feature request #2395

**Comment by yegorski** [comment]
That request looks good to me, would be a satisfactory solution.

**Comment by jlakshmipathi** [comment]
@yegorski  Are there any workarounds?

**Comment by yegorski** [comment]
No, not until https://github.com/databricks/terraform-provider-databricks/issues/2395 is implemented
